testing/yuca/ca/test_reaction_diffusion.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+
+import numpy as np
+import torch
+
+from yuca.ca.reaction_diffusion import RxnDfn
+
+class TestRxnDfn(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def test_forward(self):
+
+        rxn = RxnDfn()
+
+        x = torch.rand(1,2,64,64)
+
+        next_x = rxn(x)
+
+        self.assertEqual(x.shape, next_x.shape)
+
+    def test_update_universe(self):
+
+        rxn = RxnDfn()
+
+        x = torch.rand(1,2,64,64)
+
+        identity = rxn.id_conv(x)
+        neighborhood = rxn.neighborhood_conv(x)
+        update = rxn.update_universe(identity, neighborhood)
+        next_x = rxn(x)
+
+        sum_of_error = torch.abs(next_x - (x+rxn.dt*update)).sum()
+
+        self.assertAlmostEqual(sum_of_error, 0.0)
+
+        self.assertEqual(x.shape, next_x.shape)
+
+    def test_id_conv(self):
+
+
+        for channels in [2]:
```

Comment:
Reaction diffusion system is a less flexible framework, following the 2D Gray-Scott model. There are always 2 species (u and v) so the internal/external channels specification is not exposed to the user.

Reply:
Warning added when users try to specify channels other than 2, commit 22ac995c9a1a41b9af1dbe4dd16e0e2bd32d5a11